src/pages/api/auth-connect/send-two-fa-validation-code.ts

```diff
@@ -0,0 +1,163 @@
+import {
+  IdentityValueCantBeBlankError,
+  IdentityTypes,
+  ConnectUnreachableError,
+  sendTwoFAVerificationCode,
+  InvalidIdentityTypeError,
+} from "@fewlines/connect-management";
+import { getTracer } from "@fwl/tracing";
+import {
+  Endpoint,
+  HttpStatus,
+  getServerSideCookies,
+  setAlertMessagesCookie,
+} from "@fwl/web";
+import {
+  loggingMiddleware,
+  wrapMiddlewares,
+  tracingMiddleware,
+  errorMiddleware,
+  recoveryMiddleware,
+  rateLimitingMiddleware,
+} from "@fwl/web/dist/middlewares";
+import { NextApiRequest, NextApiResponse } from "next";
+
+import { SudoEventId } from "@src/@types/dynamo-user";
+import { Handler } from "@src/@types/handler";
+import { UserCookie } from "@src/@types/user-cookie";
+import { insertSudoEventId } from "@src/commands/insert-sudo-event-id";
+import { config } from "@src/config";
+import { NoUserFoundError } from "@src/errors";
+import { logger } from "@src/logger";
+import { authMiddleware } from "@src/middlewares/auth-middleware";
+import { sentryMiddleware } from "@src/middlewares/sentry-middleware";
+import { generateAlertMessage } from "@src/utils/generateAlertMessage";
+import { getIdentityType } from "@src/utils/get-identity-type";
+import { ERRORS_DATA, webErrorFactory } from "@src/web-errors";
+
+const handler: Handler = (request, response): Promise<void> => {
+  const webErrors = {
+    identityInputCantBeBlank: ERRORS_DATA.IDENTITY_INPUT_CANT_BE_BLANK,
+    connectUnreachable: ERRORS_DATA.CONNECT_UNREACHABLE,
+    databaseUnreachable: ERRORS_DATA.DATABASE_UNREACHABLE,
+    noUserFound: ERRORS_DATA.NO_USER_FOUND,
+    invalidIdentityType: ERRORS_DATA.INVALID_IDENTITY_TYPE,
+  };
+
+  return getTracer().span(
+    "send-two-fa-validation-code handler",
+    async (span) => {
+      const { callbackUrl, identityInput } = request.body;
+
+      if (!identityInput.value) {
+        throw webErrorFactory(webErrors.identityInputCantBeBlank);
+      }
+
+      const userCookie = await getServerSideCookies<UserCookie>(request, {
+        cookieName: "user-cookie",
+        isCookieSealed: true,
+        cookieSalt: config.cookieSalt,
+      });
+
+      if (!userCookie) {
+        response.statusCode = HttpStatus.TEMPORARY_REDIRECT;
+        response.setHeader("location", "/");
+        response.end();
+        return;
+      }
+
+      const identity = {
+        type: getIdentityType(identityInput.type),
+        value: identityInput.value,
+      };
+
+      return await sendTwoFAVerificationCode(config.managementCredentials, {
+        callbackUrl,
+        identity,
```

Comment:
`callbackUrl` can be undefined, so I would do 
```suggestion
      return await sendTwoFAVerificationCode(config.managementCredentials, {
        callbackUrl: callbackUrl || "/",
        identity,
```

Maybe we could improve this by typing as possibily `undefined` somewhere, but this is not the scope of this PR.

Reply:
We would need to do it for other flows too indeed.